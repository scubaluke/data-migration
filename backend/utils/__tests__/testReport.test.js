const { newData, oldData } = require('../testDataset.js');
const generateReport = require('../generateReport');

let result;
beforeAll(async (done) => {
  result = generateReport(newData, oldData);
  console.log('first results', result);
  done();
});
describe('when generateReport runs', () => {
  test('finds records missed in migration', () => {
    expect(result.missedRecords[0].name).toEqual('Missed in migration');
  });
  test('finds corrupted records', () => {
    expect(result.corruptedRecords[0].name).toEqual('corrupted');
    expect(result.corruptedRecords[1].email).toEqual('corruptEmail@gmail.com');
  });
  test('finds new records', () => {
    expect(result.newRecords[0].name).toEqual('Newly added data');
  });
});
