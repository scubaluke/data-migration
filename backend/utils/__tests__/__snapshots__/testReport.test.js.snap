// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when generateReport runs matches snapshot 1`] = `
Object {
  "corruptedRecords": Array [
    Object {
      "email": "myemail@gmail.com",
      "id": 1,
      "name": "corrupted",
    },
    Object {
      "email": "corruptEmail@gmail.com",
      "id": 12,
      "name": "steve",
    },
  ],
  "missedRecords": Array [
    Object {
      "email": "4someemail@gmail.com",
      "id": 14,
      "name": "Missed in migration",
    },
  ],
  "newlyCreatedRecords": Array [
    Object {
      "email": "youremail@gmail.com",
      "id": 2,
      "name": "Newly added data",
    },
  ],
}
`;
